### PR TITLE
feat(3013): Add handling for the Expect header at onPreAuth

### PR DESCRIPTION
**Pull request by foxtrot0304**
## Context

I would like to reduce unnecessary retries when uploading large files.
By using the `Expect: 100-continue` header, the content length can be checked before the body is uploaded.
Thus, the HTTP client (store-cli) can receive a 413 error earlier without unnecessary retries.

I have also submitted a related PR to `store-cli`.

https://github.com/screwdriver-cd/store-cli/pull/82

## Objective

To reduce unnecessary retries, return a 413 error earlier.

## References

[issue](https://github.com/screwdriver-cd/screwdriver/issues/3013)
[store-cli](https://github.com/screwdriver-cd/store-cli/pull/82)

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by yuichi10 — APPROVED** [review]

**Review by tkyi — DISMISSED** [review]

File: plugins/caches.js [hunk 1]
```diff
@@ -291,6 +291,28 @@ exports.plugin = {
...
+                        onPreAuth: {
+                            method: (request, h) => {
+                                const contentLength = request.headers['content-length'];
+                                const expectHeader = request.headers.expect;
+
+                                if (expectHeader && expectHeader.toLowerCase() === '100-continue') {
+                                    if (contentLength && parseInt(contentLength, 10) > options.maxByteSize) {
+                                        request.log(
+                                            'Payload content length greater than maximum allowed',
+                                            options.maxByteSize
```
Comment (s-yoshika) [inline plugins/caches.js:305]: This log seems to only output the server's max byte size. Is it necessary?

File: plugins/caches.js [hunk 2]
```diff
@@ -291,6 +291,28 @@ exports.plugin = {
...
+                                        );
+                                        throw boom.entityTooLarge();
+                                    } else {
+                                        request.raw.res.writeContinue();
```
Comment (s-yoshika) [inline plugins/caches.js:309]: [small] It looks like this line can be moved outside of the if statement.

Comment (foxtrot0304) [reply plugins/caches.js:309]: exactly ! I'll fix it.

File: plugins/caches.js (hunk 1 above)
Comment (foxtrot0304) [reply plugins/caches.js:305]: Thank you for your comment.
This log entry includes a variable. I wanted to write it in one line, but the linter gave an error, so I had to split it.
I want to log when the `maxByteSize` limit is reached. The number is just for reference.

**Review by s-yoshika — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Pull request merged**